Upstream: Yes
Source: https://chromium.googlesource.com/chromium/src/+/e98f8ef8b2f236ecbb01df8c39e6ee1c8fbe8d7d

From e98f8ef8b2f236ecbb01df8c39e6ee1c8fbe8d7d Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Thu, 13 Sep 2018 15:17:29 +0000
Subject: [PATCH] ScrollPaintPropertyNode: Rename SnapContainerData() to
 GetSnapContainerData()

GCC is stricter than clang when it comes to class members' names and how
they can change the meaning of a previously existing symbol with the same
name.

Here is a short error message:

error: changes meaning of 'SnapContainerData' from 'using SnapContainerData =
class cc::SnapContainerData' [-fpermissive]  using SnapContainerData =
cc::SnapContainerData;

Cq-Include-Trybots: luci.chromium.try:linux_layout_tests_slimming_paint_v2;master.tryserver.blink:linux_trusty_blink_rel
Change-Id: I0de5460f17b13484253e1ff0538b111c9a3b0d13
Reviewed-on: https://chromium-review.googlesource.com/1213180
Commit-Queue: Maksim Sisov <msisov@igalia.com>
Reviewed-by: Philip Rogers <pdr@chromium.org>
Cr-Commit-Position: refs/heads/master@{#591015}
---
 .../renderer/core/paint/paint_property_tree_update_tests.cc     | 2 +-
 .../platform/graphics/compositing/property_tree_manager.cc      | 2 +-
 .../platform/graphics/paint/scroll_paint_property_node.h        | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/third_party/blink/renderer/core/paint/paint_property_tree_update_tests.cc b/third_party/blink/renderer/core/paint/paint_property_tree_update_tests.cc
index 0d999d45076f3..1fe1815a5659c 100644
--- third_party/blink/renderer/core/paint/paint_property_tree_update_tests.cc
+++ third_party/blink/renderer/core/paint/paint_property_tree_update_tests.cc
@@ -1310,7 +1310,7 @@ TEST_P(PaintPropertyTreeUpdateTest, EnsureSnapContainerData) {
   GetDocument().View()->Resize(300, 300);
   GetDocument().View()->UpdateAllLifecyclePhases();
 
-  auto doc_snap_container_data = DocScroll()->SnapContainerData();
+  auto doc_snap_container_data = DocScroll()->GetSnapContainerData();
   ASSERT_TRUE(doc_snap_container_data);
   EXPECT_EQ(doc_snap_container_data->scroll_snap_type().axis, SnapAxis::kBoth);
   EXPECT_EQ(doc_snap_container_data->scroll_snap_type().strictness,
diff --git a/third_party/blink/renderer/platform/graphics/compositing/property_tree_manager.cc b/third_party/blink/renderer/platform/graphics/compositing/property_tree_manager.cc
index 745b8c8b387ea..5f87eeea94357 100644
--- third_party/blink/renderer/platform/graphics/compositing/property_tree_manager.cc
+++ third_party/blink/renderer/platform/graphics/compositing/property_tree_manager.cc
@@ -333,7 +333,7 @@ void PropertyTreeManager::CreateCompositorScrollNode(
           scroll_node->OverscrollBehaviorX()),
       static_cast<cc::OverscrollBehavior::OverscrollBehaviorType>(
           scroll_node->OverscrollBehaviorY()));
-  compositor_node.snap_container_data = scroll_node->SnapContainerData();
+  compositor_node.snap_container_data = scroll_node->GetSnapContainerData();
 
   auto compositor_element_id = scroll_node->GetCompositorElementId();
   if (compositor_element_id) {
diff --git a/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h b/third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
index e1479269ccc37..5bd7f65f10bac 100644
--- third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
+++ third_party/blink/renderer/platform/graphics/paint/scroll_paint_property_node.h
@@ -99,7 +99,7 @@ class PLATFORM_EXPORT ScrollPaintPropertyNode
     return state_.overscroll_behavior.y;
   }
 
-  base::Optional<SnapContainerData> SnapContainerData() const {
+  base::Optional<SnapContainerData> GetSnapContainerData() const {
     return state_.snap_container_data;
   }
 
